libcxx/include/__ranges/zip_view.h

```diff
@@ -468,6 +473,18 @@ inline constexpr auto zip = __zip::__fn{};
 } // namespace views
 } // namespace ranges
 
+template <class _Iter>
+  requires _Iter::__is_zip_view_iterator::value
+struct __product_iterator_traits<_Iter> {
+  static constexpr size_t __size = tuple_size<decltype(std::declval<_Iter>().__current_)>::value;
+
+  template <size_t _Nth>
+    requires(_Nth < __size)
+  _LIBCPP_HIDE_FROM_ABI static constexpr auto __get_iterator_element(_Iter __it) {
+    return std::get<_Nth>(__it.__current_);
+  }
```

Comment:
Do you intend to be making copies here? You're making a copy of the `zip_view::iterator` as an argument to the function, and you're making a copy of the sub-iterator in the return value.

This should get a test case.